# Movie Theatre

AOC Day 9 Part 1

In [15]:
input = [
    "7,1",
    "11,1",
    "11,7",
    "9,7",
    "9,5",
    "2,5",
    "2,3",
    "7,3"
]

coordinates = [
    (int(inp.split(",")[0]), int(inp.split(",")[1])) for inp in input
]

coordinates

[(7, 1), (11, 1), (11, 7), (9, 7), (9, 5), (2, 5), (2, 3), (7, 3)]

In [16]:
max_area = 0

for I in range(len(coordinates) - 1):
    for J in range(1, len(coordinates)):
        iy, ix = coordinates[I]
        jy, jx = coordinates[J]
        if ix == jx:
            # consider the unit length of a row.
            max_area = max(abs(iy - jy + 1), max_area)
        elif iy == jy:
            # consider the unit length of a column.
            max_area = max(abs(ix - iy + 1), max_area)
        else:
            max_area = max(abs(ix - jx + 1) * abs(iy - jy + 1), max_area)

print(max_area)

50


$50$ is the answer!

Let's try it for `input.txt`.

In [17]:
with open(file="input.txt") as file:
    input = [line.rstrip() for line in file]

    coordinates = [
        (int(inp.split(",")[0]), int(inp.split(",")[1])) for inp in input
    ]       

    max_area = 0

    for I in range(len(coordinates) - 1):
        for J in range(1, len(coordinates)):
            iy, ix = coordinates[I]
            jy, jx = coordinates[J]
            if ix == jx:
                # consider the unit length of a row.
                max_area = max(abs(iy - jy + 1), max_area)
            elif iy == jy:
                # consider the unit length of a column.
                max_area = max(abs(ix - iy + 1), max_area)
            else:
                max_area = max(abs(ix - jx + 1) * abs(iy - jy + 1), max_area)

    print(max_area)

4781235324


$4781235324$ is the correct answer! 

That was super easy, so Part 2 must be the real deal.

---

### Part II